biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/DownloadChemCompProvider.java

```diff
@@ -166,62 +166,57 @@ private void split() throws IOException {
 		int counter = 0;
 		InputStreamProvider prov = new InputStreamProvider();
 
-		InputStream inStream = prov.getInputStream(f);
-
-		BufferedReader buf = new BufferedReader (new InputStreamReader (inStream));
+		try( BufferedReader buf = new BufferedReader (new InputStreamReader (prov.getInputStream(f)));
+				) {
+			String line = null;
+			line = buf.readLine ();
+			StringWriter writer = new StringWriter();
```

Comment:
Nice! I didn't know you could do this kind of thing with Java 7. This is the reference for those that don't know (like me!): https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html